lib/sectorbuilder/sectorbuilder_test.go

```diff
@@ -117,18 +148,55 @@ func TestSealAndVerify(t *testing.T) {
 		t.Fatalf("%+v", err)
 	}
 
-	postProof, err = sb.GeneratePoSt(ssi, cSeed, []uint64{})
-	if err != nil {
+	s.post(t, sb)
+}
+
+func TestSealAndVerify2(t *testing.T) {
+	//t.Skip("this is slow")
+	os.Setenv("BELLMAN_NO_GPU", "1")
+	os.Setenv("RUST_LOG", "info")
+
+	build.SectorSizes = []uint64{sectorSize}
+
+	if err := build.GetParams(true, true); err != nil {
 		t.Fatalf("%+v", err)
 	}
 
-	ok, err = sectorbuilder.VerifyPost(context.TODO(), sb.SectorSize(), ssi, cSeed, postProof, []uint64{})
+	ds := datastore.NewMapDatastore()
+
+	dir, err := ioutil.TempDir("", "sbtest")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sb, err := sectorbuilder.TempSectorbuilderDir(dir, sectorSize, ds)
 	if err != nil {
 		t.Fatalf("%+v", err)
 	}
-	if !ok {
-		t.Fatal("bad post")
+	cleanup := func() {
+		sb.Destroy()
+		if err := os.RemoveAll(dir); err != nil {
+			t.Error(err)
+		}
 	}
+
+	defer cleanup()
+
+	var wg sync.WaitGroup
+
+	s1 := seal{}
+	s2 := seal{}
+
+	wg.Add(2)
+	go s1.precommit(t, sb, 1, wg.Done)
+	time.Sleep(100 * time.Millisecond)
```

Comment:
why sleep this here?

Reply:
The issue I had manifested mostly with slightly offset precommit call

Reply:
This is still a race, it may lead to flaky tests in future. How about doing the `AcquireSectorID` in test body instead of goroutine?